Busy / error improvements

parente opened the pull request:
Show busy bar during kernel start. Show error indicator when kernel fails to start.

Fixes #269

jhpedemonte commented on client/js/dashboard.js:
```diff
@@ -108,6 +108,8 @@ if (Element && !Element.prototype.matches) {
             ErrorIndicator.show();
         });
+    }).catch(function(e) {
+        ErrorIndicator.show();
```
I think we can make this cleaner:

1. Return the promise from `widgetsReady.then()` (line 67).
2. Remove the catch for that line, lines 107-110.
3. Add `console.error(err)` to your new code.

This way, there's only one error handler rather than 2.

parente replied:
What about the console.error happening down in kernel.js? Was thinking it's kinda weird that we do some logging in one place and other logging/error handling in another? Should that move up too?

jhpedemonte replied:
Oh, yes, I see. So a unified error handler could get errors from (1) `Kernel.start` (in which case it has been logged) ; or (2) `widgetsReady` (where it has not been logged).

I think it makes the most sense to remove logging from `Kernel.start`, since it is also throwing the error. It should probably just create a new error with the text `Failed to create kernel`, wrapping the incoming error. Make sense?

jhpedemonte commented on client/js/kernel.js:
```diff
@@ -43,17 +54,14 @@ var Services = require('jupyter-js-services');
-                            // Prevent progress animation when hidden by removing 'active' class.
-                            .find('.progress-bar')
-                                .toggleClass('active', isBusy);
+                        var isBusy = status === Services.KernelStatus.Busy;
```
I thought `Services.KernelStatus` doesn't exist anymore in latest **jupyter-js-services**. I had previously changed all references to use string values instead.

parente replied:
Errr. That's a bad rebase. I'll fix.

parente commented:
@jhpedemonte I think I addressed your comments in the way you were saying. I tested the error handling by killing the KG and watching for the indicator. I tried to test against the scala widgets notebook, but it seems to have stopped working since the PR this morning. (I'm guessing it's the external declarativewidgets.jar that's not getting pulled now for some external reason ...)

jhpedemonte commented on client/js/kernel.js:
```diff
@@ -44,18 +55,14 @@ var Services = require('jupyter-js-services');
-                console.error('failed to create kernel', e);
-                throw e;
+                showBusyIndicator(false);
+                throw new Error('Failed to create kernel');
```
I would change this to:

`throw new Error('Failed to create kernel: ' + e.message)`

This will provide more information.

jhpedemonte commented:
Just one minor nit. Otherwise, it's good to merge.

parente commented:
Fixed nit. Merge away when green.
